python/ray/train/lightning/_lightning_utils.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from pytorch_lightning.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
+import torch
+from typing import Any, Dict, Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.strategies import DDPStrategy
+from pytorch_lightning.plugins.environments import LightningEnvironment
+
+import ray
+from ray.air import session
+
+from torch.utils.data import IterableDataset, DataLoader
+from ray.data.dataset import Dataset
+
+logger = logging.getLogger(__name__)
+
+
+class RayDDPStrategy(DDPStrategy):
+    """Subclass of DDPStrategy to ensure compatibility with Ray orchestration."""
+
+    @property
+    def root_device(self) -> torch.device:
+        return ray.train.torch.get_device()
+
+    @property
+    def distributed_sampler_kwargs(self) -> Dict[str, Any]:
+        return dict(
+            num_replicas=self.world_size,
+            rank=self.global_rank,
+        )
+
+
+class RayEnvironment(LightningEnvironment):
+    """Setup Lightning DDP training environment for Ray cluster."""
+
+    def world_size(self) -> int:
+        return session.get_world_size()
+
+    def global_rank(self) -> int:
+        return session.get_world_rank()
+
+    def local_rank(self) -> int:
+        return session.get_local_rank()
+
+    def node_rank(self) -> int:
+        return session.get_node_rank()
+
+    def set_world_size(self, size: int) -> None:
+        self._world_size = session.get_world_size()
+
+    def set_global_rank(self, rank: int) -> None:
+        self._global_rank = session.get_world_rank()
+        rank_zero_only.rank = rank
+
+    def teardown(self):
+        pass
+
+
+class RayIterableDataset(IterableDataset):
+    def __init__(self, dataset: "Dataset", config: Dict[str, Any]) -> None:
+        super().__init__()
+        self.dataset = dataset
+        self.config = config
+
+    def __iter__(self):
+        return self.dataset.iter_torch_batches(**self.config)
+
+
+class RayDataModule(pl.LightningDataModule):
+    def __init__(
+        self,
+        dataset_iter_config: Dict[str, Any],
+        train_dataset: "Dataset",
+        val_dataset: Optional["Dataset"] = None,
+    ) -> None:
+        super().__init__()
+
+        if not dataset_iter_config:
+            raise RuntimeError(
+                "To use Ray Datasets with LightningTrainer, you must specify "
+                " `datasets_iter_config` in LightningConfig!"
+            )
+
+        def _train_dataloader() -> TRAIN_DATALOADERS:
```

Comment:
It seems that they changed this variable to `Any` only 3 weeks ago. [Link](https://github.com/Lightning-AI/lightning/commit/ec4f592ecfe238edd83185f6c6905fb1e2406d61#diff-ff8ae90d0073754ac92dc085581e5322c456580894c217f647deea13417fd3d3L35)

Previously it was:
```
TRAIN_DATALOADERS = Union[
    DataLoader,
    Sequence[DataLoader],
    Sequence[Sequence[DataLoader]],
    Sequence[Dict[str, DataLoader]],
    Dict[str, DataLoader],
    Dict[str, Dict[str, DataLoader]],
    Dict[str, Sequence[DataLoader]],
]
EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
```

But yeah, since they already changed it to `Any`, it should be safe to declare `DataLoader` as return type.